### ISSUE TITLE
Possible regression in proxy use for update checks in 3.4.0-betaX appimages.

### ISSUE THREAD

**Issue by zedgoat** [standardnotes/desktop#535]
# Description

In previous revision of standard notes, 3.3.5 installed via [AUR](https://aur.archlinux.org/packages/standardnotes-desktop/), SN has leveraged the proxy environment settings (`HTTP_PROXY` and `HTTPS_PROXY`) for application update checks on my workstation (Manjaro 5.6.7-1-MANJARO).

On download of [3.4.0-beta3](https://github.com/standardnotes/desktop/releases/tag/v3.4.0-beta3), the application loads existing local cache and content OK, but sync and update checks fail with:

![image](https://user-images.githubusercontent.com/19355901/80672701-bc0d6500-8af0-11ea-8067-7fdb06cc4aa7.png)

The omitted hostname is an upstream intercepting proxy that applications hit when they ignore the local proxy configuration.

Executing the app from the terminal:

![image](https://user-images.githubusercontent.com/19355901/80672935-51105e00-8af1-11ea-931b-76ff00cea4e4.png)

There isn't corresponding events in the Developer Tools in SN.

# Testing

Had a naive attempt at executing SN with `ELECTRON_GET_USE_PROXY=1 GLOBAL_AGENT_HTTP_PROXY=http://127.0.0.1:8888 GLOBAL_AGENT_HTTPS_PROXY=http://127.0.0.1:8888 ./Standard.Notes-3.4.0-beta3.AppImage` after reading the [Node 10 docs](https://github.com/gajus/global-agent/blob/v2.1.5/README.md#environment-variables), in response to the note on the [installation page](https://www.electronjs.org/docs/tutorial/installation) about Node 10 onwards having different proxy handling requirements. No change, but this could also be an idiots attempt at a fix ;)

Testing of `beta2` shows the same issue.

# Impact

For me not much: under normal use I rely on AUR to get SN updates so don't depend on the internal update mechanism, and disabling automatic update suppresses the prompt. But for other's depending on the appimage update behind a proxy (probably a small user pool), it could have impact.

**Comment by baptiste-grob** [comment]
Wow, very good catch. Thank you! I'll look into it.

**Comment by mobitar** [comment]
Thanks for the detailed ticket @zedgoat. Does this relate only to update checks? What about downloading extensions, does that work? The desktop app will attempt to download any installed extensions for offline use. Although I suppose in previous forms failure here would be silent, and the app would fallback to hosted extensions. Could you inspect the iframe for editors in previous versions to see if the src is a file:// path for https:// path? And compare that to beta behavior?

It was my understanding that the desktop app never quite respected proxy settings, as there have been a few issues over the years where users explicitly requested this feature.

**Comment by zedgoat** [comment]
@baptiste-grob no worries.

@mobitar Good point. I tried purging `~.config/Standard Notes` and restarting: login, sync, notes are OK, update checks continue to fail.

## "Manual" SN update check, grab update URL from devtools.

From the repo `standardnotes/desktop`, in the update manager code: `desktop/app/javascripts/main/updateManager.ts`

`curl https://standardnotes.org/desktop/latest.json`, OK. `wget` of the payloads at that location also OK.

Firing a `fetch('https://standardnotes.org/desktop/latest.json').then(res => res.json()).then(console.log)` from within the Developer Tools brings the right content back too. Not sure if this means squat, not a js / ts / electron / node guy at all.

Curious, there's a difference between "Check for Update via Electron" vs "SN stuff that runs _inside of_ electron"? It only seems to be `Check for Update` that is now tripping up on proxy config.

## Misc other testing

My futile efforts, but might help other people chase this:

### Node / electron proxy config.

Assuming latest Electron / v8 tweaked something that previously worked as a happy-accident. Digging around based on:

```
getaddrinfo ENOTFOUND
GetAddrInfoReqWrap.onlookup
dns.js:60:26
```

Not much in the source that gives it away. Playing around with flags explicity (`--http-proxy` and `--https-proxy`, or `--proxy-server`, as per [docs](https://www.electronjs.org/docs/api/command-line-switches)) has no effect.

### `proxychains-ng` test

Tried running under proxychains to force it through a proxy, but proxychains segfaults when calling SN.

### Proxy formatting issues in Gnome.

Previous version of Gnome proxy config would try and be clever with URL prefix in their proxy setup, which would cause some apps to break (depending on if they expected / didn't expect the prefix). Prefix proxy with `http://` instead of naked URL: still fails.

**Comment by baptiste-grob** [comment]
Thank you very much for your help! We've implemented a fix that will land in the soon to be released version 3.4 of the app :)

**Comment by zedgoat** [comment]
@baptiste-grob awesome, thank you for the effort for something that likely only impacts a small number of users!
